chore: fulu nits

matthewkeil opened the pull request:
**Motivation**

Addresses peerDAS PR comments
https://github.com/ChainSafe/lodestar/pull/6353#discussion_r2262381356
https://github.com/ChainSafe/lodestar/pull/6353#discussion_r2262749064
https://github.com/ChainSafe/lodestar/pull/6353#discussion_r2262760646
https://github.com/ChainSafe/lodestar/pull/6353#discussion_r2262765986
https://github.com/ChainSafe/lodestar/pull/6353#discussion_r2262806254
https://github.com/ChainSafe/lodestar/pull/6353#discussion_r2262823140
https://github.com/ChainSafe/lodestar/pull/6353#discussion_r2263489898
https://github.com/ChainSafe/lodestar/pull/6353#discussion_r2263499795
https://github.com/ChainSafe/lodestar/pull/6353#discussion_r2263535882
https://github.com/ChainSafe/lodestar/pull/6353#discussion_r2263521333

matthewkeil committed:
fix: use permalink

matthewkeil committed:
fix: mock return type

matthewkeil committed:
fix: wow that was a nit nflaig!!!  love you dude!

matthewkeil committed:
refactor: use type WithOptionalBytes

matthewkeil committed:
fix: revert to altair.Metadata

matthewkeil committed:
fix: change forkName argument index in assertPeerRelevance

matthewkeil committed:
fix: use MAX_REQUEST_DATA_COLUMN_SIDECARS

matthewkeil committed:
fix: unpin lightclient ResponseBodyByMethod from altair

matthewkeil committed:
fix: make metrics collection optional in SyncChain

nflaig commented on packages/beacon-node/src/chain/validation/dataColumnSidecar.ts:
```diff
@@ -247,7 +247,7 @@ export async function validateDataColumnsSidecars(
 /**
  * SPEC FUNCTION
- * https://github.com/ethereum/consensus-specs/blob/dev/specs/fulu/p2p-interface.md#verify_data_column_sidecar
+ * https://github.com/ethereum/consensus-specs/blob/ad36024441cf910d428d03f87f331fbbd2b3e5f1/specs/fulu/p2p-interface.md#L82
```
there are a lot more links that reference `dev`

matthewkeil replied:
Actually just checked... There are a lot.  That should be in a separate PR to `unstable`

<img width="734" height="1659" alt="Screenshot 2025-08-09 at 6 31 02 AM" src="https://github.com/user-attachments/assets/6da5f707-371e-4691-af02-046a6bbfd06e" />

nflaig replied:
yeah that's bad enough, could at least update references you find when searching for `dev/specs/fulu`

matthewkeil replied:
would prefer to do this under separate PR.  made an issue for it
https://github.com/ChainSafe/lodestar/issues/8171

matthewkeil committed:
refactor: rename batch.isPostFulu

matthewkeil committed:
fix: bad import

nflaig commented on packages/beacon-node/src/chain/validation/dataColumnSidecar.ts:
```diff
@@ -247,7 +247,7 @@ export async function validateDataColumnsSidecars(
 /**
  * SPEC FUNCTION
- * https://github.com/ethereum/consensus-specs/blob/dev/specs/fulu/p2p-interface.md#verify_data_column_sidecar
+ * https://github.com/ethereum/consensus-specs/blob/ad36024441cf910d428d03f87f331fbbd2b3e5f1/specs/fulu/p2p-interface.md#L82
```
just noticed the line reference `#L82` does not work for me, requires the `plain` flag in the url so the markdown file is not rendered

matthewkeil replied:
updated

nflaig committed:
Use tag in spec references

We can later consider updating this to v1.6.0, it's cleaner than having a commit hash which is too long

nflaig committed:
Fix metric type inference, remove unused this.metrics

nflaig committed:
Remove unused imports

wemeetagain commented:
:tada: This PR is included in [v1.34.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.34.0) :tada:
